src/Knp/Menu/Matcher/Voter/RouteVoter.php

```diff
@@ -83,7 +83,7 @@ private function isMatchingRoute(array $testedRoute)
         $routeParameters = $this->request->attributes->get('_route_params', array());
 
         foreach ($testedRoute['parameters'] as $name => $value) {
-            if (!isset($routeParameters[$name]) || (string) $routeParameters[$name] !== (string) $value) {
+            if (!isset($routeParameters[$name]) || $routeParameters[$name] !== (string) $value) {
```

Comment:
@stof I dumped the parameters [right after what I think is what you are referring to](https://github.com/symfony/symfony/blob/2.8/src/Symfony/Component/HttpKernel/EventListener/RouterListener.php#L158) in the router listener, and I get an integer, are you sure it can't be an integer ?